### PR TITLE
A11y popup menu

### LINKED ISSUE #871: Make popup menu WCGA 2.1 compliant
### Is your feature request related to a problem? Please describe.

Currently the popup menu is a keyboard trap that is insufficiently indicated to assistive technology (screen readers), it violates [`no-keyboard-trap`](https://www.w3.org/WAI/WCAG22/quickref/#no-keyboard-trap).

### Describe the solution you'd like

The popup editor is clearly indicated as a keyboard trap with `ESC` (cancel) or `ENTER` (to select action) being the two promoted navigation actions.

### Describe alternatives you've considered

None.

### Additional context

Found via https://github.com/camunda/web-modeler/issues/7929#issuecomment-2003338026.

### PR DESCRIPTION
**Pull request by nikku**
#### Firefox

capture 5fPyRh_optimized

#### Chrome

capture 5w5PjG_optimized

---

This PR improves the accessibility of the popup menu:

* Ensures all items are keyboard accessible
* Ensures that existing TAB and arrow key navigation continues to work
* Ensures that `SPACE` and `ENTER` can be used as expected to toggle features
* Ensure that outside click is no longer trapped
* Use browser native outline for keyboard focus state

---

Additional notes:

* I decided that tabbing through the whole component (all entries) shall be possible
* I decided against making this a partial select/fake drop down from aria perspective, as it does not work in scenarios without search

----

Can be tried out using the following command on your local machine:

```
npx @bpmn-io/sr bpmn-io/bpmn-js -l bpmn-io/diagram-js#a11y-popup-menu
```

----

Closes #871

### CODE REVIEW

**Comment by nikku** [comment]
To be considered: Do not [browser focus existing element](https://github.com/bpmn-io/diagram-js/pull/874/commits/4a72c06a6ede8efaffdaf95b4542884d8e26b2c7), but only do so on tab navigation:

capture r7pQV7_optimized

**Comment by nikku** [comment]
As per discussion discarded the automatic keyboard focus:

* Ensure that resonable item is intially focused

---

capture 4dT3A2_optimized

**Comment by barmac** [comment]
It looks good on Chrome, but there are some problems on Safari:

https://github.com/bpmn-io/diagram-js/assets/28307541/7c9ebc06-f333-462e-9e6c-e48b76ceae91

* The outline is cut-off.
* I cannot focus headers buttons with keyboard.
* The menu is blurred when zoomed in o.O (out of scope).

**Comment by barmac** [comment]
This is Safari 17.0

**Comment by barmac** [comment]
On Firefox, the outline is also cut off, but at least I can focus the header buttons, and the menu is never blurred:

image

**Comment by nikku** [comment]
So we have to decide if we invent a completely new outline, standardized across all browsers.

I'd keep that as a follow-up activity.

**Comment by barmac** [comment]
I'm OK with outline cut-off but it would be great if the focus worked correctly. I will check in the dev tools what is happening there.

**Comment by barmac** [comment]
So via TAB I cannot reach the header modules, and after I TAB out, it focusses the body.

File: lib/features/popup-menu/PopupMenuItem.js [hunk 1]
```diff
@@ -33,10 +33,14 @@ export default function PopupMenuItem(props) {
...
       data-id=${ entry.id }
       title=${ entry.title || entry.label }
+      tabIndex="0"
       onClick=${ onAction }
+      onFocus=${ onMouseEnter }
+      onBlur=${ onMouseLeave }
       onMouseEnter=${ onMouseEnter }
       onMouseLeave=${ onMouseLeave }
       onDragStart=${ (event) => onAction(event, entry, 'dragstart') }
+      aria-role="button"
```
Comment (barmac) [inline lib/features/popup-menu/PopupMenuItem.js:43]: If the items serve as buttons now, why don't we make them actual buttons? Spans make sense if we want to keep the select-option-like behavior but we decided against this as I understand.

File: lib/features/popup-menu/PopupMenuComponent.js [hunk 2]
```diff
@@ -296,26 +260,58 @@ function PopupMenuWrapper(props) {
...
-    popupRef.current && popupRef.current.focus();
+    const popupEl = popupRef.current;
+
+    if (!popupEl) {
+      return;
+    }
+
+    const inputEl = popupEl.querySelector('input');
+
+    (inputEl || popupEl).focus();
```
Comment (barmac) [inline lib/features/popup-menu/PopupMenuComponent.js:273]: I'd rather implement this via `autofocus` attribute.

**Comment by barmac** [comment]
OK it might be that I don't know my tool. I can tab to the header buttons via Option+TAB: https://stackoverflow.com/questions/1848390/safari-ignoring-tabindex

**Review by barmac — APPROVED** [review]
LGTM

**Pull request merged**